src/impl/core/dappletContext.ts

```diff
@@ -12,23 +13,19 @@ import { GithubDappletProvider } from '../providers/githubDappletProvider';
 // instantated once on Wallet start
 export class DappletContext {
     private _dappletProvider: DappletProvider;
+    private _featuresRegistry: FeaturesRegistry;
 
-    constructor(config: ContextConfig) {
-        switch (config.source) {
-            case "github":
-                this._dappletProvider = new GithubDappletProvider();                
-                break;
-        
-            default:
-                this._dappletProvider = new GithubDappletProvider();    
-                break;
-        }
+    constructor(config?: ContextConfig) {
+        if (!config) config = DEFAULT_CONFIG;
+
+        this._dappletProvider = config.provider;
+        this._featuresRegistry = config.features;
     }
 
     //typeSupport: Map<PID,(rawData:string,type:PID)=>typedValue> = new Map
 
     async processRequest(request: DappletRequest) {
-        const engine = new DappletEngine(request);
+        const engine = new DappletEngine(request, this._featuresRegistry);
         await engine.load(this._dappletProvider);
         if (!engine.validate()) throw new Error("Invalid dapplet");
```

Comment:
I would suggest;
`        const engine = new DappletEngine(this._context);
        await engine.process(request);  //implies loading and verification and fails on errors
`